src/components/grid-layout/cards/diagrams/singleLineDiagram/hooks/use-one-bus-shortcircuit-analysis-loader.tsx

```diff
@@ -40,38 +39,52 @@ type oneBusShortcircuitAnalysisLoader = [ReactElement, boolean, () => void, () =
  * A hook that handles the logic behind the diagram one bus shortcircuit analysis loader
  *
  * @param {string} diagramId - Identifier for the diagram which launched the computation
- * @param {UUID} nodeId - Identifier for the node which launched the computation
- * @param rootNetworkUuid
  *
  * @returns {oneBusShortcircuitAnalysisLoader} array which contains the controls necessary for the one bus
  * shortcircuit analysis loader. It also comes with a boolean to check if the loader needs to be displayed
  * and the message to display for the UI
  */
-export function useOneBusShortcircuitAnalysisLoader(
-    diagramId: string,
-    nodeId: UUID,
-    rootNetworkUuid: UUID
-): oneBusShortcircuitAnalysisLoader {
+export function useOneBusShortcircuitAnalysisLoader(diagramId: string): oneBusShortcircuitAnalysisLoader {
+    const studyUuid = useSelector((state: AppState) => state.studyUuid);
+    const currentNode = useSelector((state: AppState) => state.currentTreeNode);
+    const rootNetworkUuid = useSelector((state: AppState) => state.currentRootNetworkUuid);
+
     const studyUpdatedForce = useSelector((state: AppState) => state.studyUpdated);
     const oneBusShortCircuitAnalysisDiagram = useSelector((state: AppState) => state.oneBusShortCircuitAnalysisDiagram);
 
     const dispatch = useDispatch<AppDispatch>();
     const intl = useIntl();
 
     const displayOneBusShortcircuitAnalysisLoader = useCallback(() => {
-        dispatch(setOneBusShortcircuitAnalysisDiagram(diagramId, nodeId));
-    }, [nodeId, diagramId, dispatch]);
+        if (!studyUuid || !currentNode?.id || !rootNetworkUuid) {
+            return;
+        }
+        dispatch(setOneBusShortcircuitAnalysisDiagram(diagramId, currentNode?.id, rootNetworkUuid));
```

Comment:
I think node id is not necessary !

Reply:
as discussed I will add `studyUuid` in the context stored to know on which diagram to show the loader.